### PR TITLE
add sources, targets, multi relationship_types/datasets for relationships, remove use of deprecated params / 1.4.13

### PR DESCRIPTION
**Pull request by sanderland**

### CODE REVIEW

File: cognite/client/_api/relationships.py [hunk 1]
```diff
@@ -13,20 +13,68 @@ def __init__(self, *args, **kwargs):
...
+        data_set: Optional[Union[str, List[str]]] = None,
+        relationship_type: Optional[Union[str, List[str]]] = None,
+    ):
+        if sources and (source_resource or source_resource_id):
+            raise ValueError("Can not set both sources and source_resource/source_resource_id.")
+        if targets and (target_resource_id or target_resource):
+            raise ValueError("Can not set both targets and target_resource/target_resource_id.")
+        if relationship_type is not None and not isinstance(relationship_type, list):
+            relationship_type = [relationship_type]
+        if data_set is not None and not isinstance(data_set, list):
```
Comment (andreavs) [inline cognite/client/_api/relationships.py]: arguably just if isinstance(data_set, str)

File: cognite/client/_api/relationships.py [hunk 2]
```diff
@@ -37,34 +85,38 @@ def __call__(
...
             target_resource_id (str): Resource ID of the target node.
+            targets (List[Dict[str, Any]]): List of multiple targets in the format `[{"resourceId":externalId,"resource":"Asset"},...]`
             start_time (Dict[str, Any]): Range to filter the field for. (inclusive)
             end_time (Dict[str, Any]): Range to filter the field for. (inclusive)
             confidence (Dict[str, Any]): Range to filter the field for. (inclusive)
             last_updated_time (Dict[str, Any]): Range to filter the field for. (inclusive)
             created_time (Dict[str, Any]): Range to filter the field for. (inclusive)
-            data_set (str): String describing the source system storing or generating the relationship.
-            relationship_type (str): Type of the relationship in order to distinguish between different relationships. In general relationship types should reflect references as they are expressed in natural sentences.
+            data_set (Union[str,List[str]): Filter on a single dataset or one of a given list of datasets.
```
Comment (andreavs) [inline cognite/client/_api/relationships.py]: i would say  "filter on any of a given list of dataSets"

File: cognite/client/_api/relationships.py [hunk 3]
```diff
@@ -37,34 +85,38 @@ def __call__(
...
+            relationship_type (Union[str,List[str]):  Filter on one of these relationship types.
```
Comment (andreavs) [inline cognite/client/_api/relationships.py]: same

File: cognite/client/_api/relationships.py [hunk 4]
```diff
@@ -120,31 +172,35 @@ def list(
...
             target_resource_id (str): Resource ID of the target node.
+            targets (List[Dict[str, Any]]): List of multiple targets in the format `[{"resourceId":externalId,"resource":"Asset"},...]`
             start_time (Dict[str, Any]): Range to filter the field for. (inclusive)
             end_time (Dict[str, Any]): Range to filter the field for. (inclusive)
             confidence (Dict[str, Any]): Range to filter the field for. (inclusive)
             last_updated_time (Dict[str, Any]): Range to filter the field for. (inclusive)
             created_time (Dict[str, Any]): Range to filter the field for. (inclusive)
-            data_set (str): String describing the source system storing or generating the relationship.
-            relationship_type (str): Type of the relationship in order to distinguish between different relationships. In general relationship types should reflect references as they are expressed in natural sentences.
+            data_set (Union[str,List[str]): Filter on a single dataset or one of a given list of datasets.
```
Comment (andreavs) [inline cognite/client/_api/relationships.py]: any of

File: cognite/client/_api/relationships.py [hunk 5]
```diff
@@ -120,31 +172,35 @@ def list(
...
+            relationship_type (Union[str,List[str]):  Filter on one of these relationship types.
```
Comment (andreavs) [inline cognite/client/_api/relationships.py]: any of

File: cognite/client/data_classes/relationships.py [hunk 6]
```diff
@@ -78,47 +78,59 @@ def _resolve_target(target):
 
 # GenClass: relationshipsAdvancedListRequest.filter
 class RelationshipFilter(CogniteFilter):
-    """Filter on relationshisp with exact match.
+    """Filter on relationships with exact match. Multiple filter elments in one property, e.g. `dataSets: [ "a", "b" ]`, will return all relationships that are either in dataSet `a` or in dataSet `b`. Filters in multiple properties will return the relationships that match all criteria. Filters on a `resourceId` without a `resource` (type) in sources and targets will return relationships that match the resourceId and match any resource type.
```
Comment (andreavs) [inline cognite/client/data_classes/relationships.py]: arguably "...will return all relationships where the dataSet field is either `a` or `b`"

Comment (andreavs) [reply cognite/client/data_classes/relationships.py]: although this field needs to die in favor of proper dataSet support

File: tests/tests_integration/test_api/test_events.py [hunk 7]
```diff
@@ -64,7 +64,7 @@ def test_list(self, post_spy):
 
     def test_partitioned_list(self):
         # stop race conditions by cutting off max created time
-        maxtime = utils.timestamp_to_ms(datetime(2019, 6, 20, 17, 30))
+        maxtime = utils.timestamp_to_ms(datetime(2019, 12, 25, 17, 30))
```
Comment (andreavs) [inline tests/tests_integration/test_api/test_events.py]: no idea why this was changed in this PR but it probably doesn't break anything

File: CHANGELOG.md [hunk 8]
```diff
@@ -21,6 +21,9 @@ Changes are grouped as follows
 
 ## Unreleased
 
+### Added
+- Relationship list supports multiple sources and targets
```
Comment (andreavs) [inline CHANGELOG.md]: also relationshipTypes and dataSets?

File: cognite/client/data_classes/relationships.py (hunk 6 above)
Comment (sanderland) [reply cognite/client/data_classes/relationships.py]: these are auto generated from your specs :)

File: tests/tests_integration/test_api/test_events.py (hunk 7 above)
Comment (sanderland) [reply tests/tests_integration/test_api/test_events.py]: this is unrelated :)

**Pull request merged**